packages/opencensus-exporter-trace-zipkin/src/zipkin.ts

```diff
@@ -0,0 +1,178 @@
+    this.buffer = new classes.ExporterBuffer(this, options);
+    this.logger = options.logger || logger.logger();
+  }
+
+  /**
+   * Is called whenever a span is ended.
+   * @param root the ended span
+   */
+  onEndSpan(root: types.RootSpan) {
+    this.buffer.addToBuffer(root);
+  }
+
+  /**
+   * Send a trace to zipkin service
+   * @param zipkinTraces Trace translated to Zipkin Service
+   */
+  private async sendTraces(zipkinTraces: TranslatedSpan[]) {
+    /** Request options */
+    const options = {
+      hostname: this.zipkinUrl.hostname,
+      port: this.zipkinUrl.port,
+      path: this.zipkinUrl.path,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    };
+
+    return new Promise((resolve, reject) => {
+      try {
+        /** Request object */
+        const req = http.request(options, (res) => {
+          res.on('data', (chunk) => {});
+          // Resolve on end
+          res.on('end', () => {
+            resolve(
+                {statusCode: res.statusCode, statusMessage: res.statusMessage});
+          });
+        });
+
+        /** Request error event */
+        req.on('error', (e) => {
+          reject({
+            statusCode: 500,
+            statusMessage: `Problem with request: ${e.message}`
+          });
+        });
+
+        /** Request body */
+        const spansJson: string[] =
+            zipkinTraces.map((span) => JSON.stringify(span));
+        spansJson.join('');
+        const outputJson = `[${spansJson}]`;
+        this.logger.debug('Zipkins span list Json: %s', outputJson);
+
+        // Sendind the request
+        req.write(outputJson, 'utf8');
+        req.end();
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  /**
+   * Mount a list (array) of spans translated to Zipkin format
+   * @param rootSpan Trace to be translated
+   */
+  private mountSpanList(spanList: TranslatedSpan[], rootSpan: types.RootSpan):
+      TranslatedSpan[] {
+    /** RootSpan data */
+    const spanRoot = this.translateSpan(rootSpan);
+    spanList.push(spanRoot);
+
+    // Builds spans data
+    for (const span of rootSpan.spans) {
+      spanList.push(this.translateSpan(span, rootSpan));
+    }
+
+    return spanList;
+  }
+
+  /**
+   * Translate OpenSensus Span to Zipkin format
+   * @param span Span to be translated
+   * @param rootSpan Only necessary if the span has rootSpan
+   */
+  private translateSpan(
+      span: types.Span|types.RootSpan,
+      rootSpan?: types.RootSpan): TranslatedSpan {
+    const spanTraslated = {
+      traceId: span.traceId,
+      name: span.name,
+      id: span.id,
+      kind: 'SERVER',
+      timestamp: (span.startTime.getTime() * 1000).toFixed(),
+      duration: (span.duration * 1000).toFixed(),
+      debug: true,
+      shared: true,
+      localEndpoint: {serviceName: this.serviceName}
+    } as TranslatedSpan;
+
+    if (rootSpan) {
+      spanTraslated.parentId = rootSpan.id;
```

Comment:
Doesn't `span` expose a `parentSpanId` field? Is it possible to use that?

Reply:
Ok. Done.